source/system/Array.h

```diff
@@ -1,5 +1,5 @@
-#ifndef OOC_LANG_ARRAY_H
-#define OOC_LANG_ARRAY_H
+#ifndef ARRAY_H_
```

Comment:
I'd leave the longer macro guard here, `ARRAY_H_` seems not unique enough to me.